pkg/preference/implem.go

```diff
@@ -133,6 +133,19 @@ func newPreferenceInfo() (*preferenceInfo, error) {
 		return nil, err
 	}
 
+	regList := []Registry{}
+	if c.OdoSettings.RegistryList != nil {
+		regList = *c.OdoSettings.RegistryList
+	}
+
+	i := 0
+	j := len(regList) - 1
+	for i < j {
+		regList[i], regList[j] = regList[j], regList[i]
+		i++
+		j--
+	}
+
```

Comment:
@anandrkskd Now that the Preference Client implementation does the reverse operation (which is a good thing), I think there is one side effect, which we should consider, IMO.
In [`pkg/registry/registry.go`](https://github.com/redhat-developer/odo/blob/main/pkg/registry/registry.go#L64-L65), the registry list is also iterated from the end, which again returns the list in the wrong order, for example when running `odo registry` or even when fetching a new Devfile with `odo init` :)

<details>
<summary>Examples</summary>

```sh
# Correct order: StagingRegistry comes first. Same with JSON output.
$ odo preference view

...
Devfile registries:
 NAME                    URL                                SECURE 
 StagingRegistry         https://registry.stage.devfile.io  No     
 DefaultDevfileRegistry  https://registry.devfile.io        No     
```

```sh
# Issue 1: For each Devfile name, StagingRegistry should be listed first
$ odo registry

...
NAME                          REGISTRY                DESCRIPTION                                 
 dotnet50                      DefaultDevfileRegistry  Stack with .NET 5.0                         
 dotnet50                      StagingRegistry         Stack with .NET 5.0                         
 dotnet60                      DefaultDevfileRegistry  Stack with .NET 6.0                         
 dotnet60                      StagingRegistry         Stack with .NET 6.0                         
 dotnetcore31                  DefaultDevfileRegistry  Stack with .NET Core 3.1                    
 dotnetcore31                  StagingRegistry         Stack with .NET Core 3.1                    
 go                            DefaultDevfileRegistry  Go is an open source programming languag... 
 go                            StagingRegistry         Go is an open source programming language... 
```

```sh
# Issue 2: Devfile should have been selected from "StagingRegistry"
$ odo init           
  __
 /  \__     Initializing a new component
 \__/  \    Files: Source code detected, a Devfile will be determined based upon source code autodetection
 /  \__/    odo version: v3.2.0
 \__/

Interactive mode enabled, please answer the following questions:
Based on the files in the current directory odo detected
Language: Java
Project type: Maven
The devfile "java-maven" from the registry "DefaultDevfileRegistry" will be downloaded.
? Is this correct? (Y/n) 
```

</details>


So I think we need to change [`pkg/registry/registry.go`](https://github.com/redhat-developer/odo/blob/main/pkg/registry/registry.go#L64-L65) to rely on the order returned by the Preference Client.

Reply:
And we should also make sure that there are no other places where the list returned by the Preference Client is iterated in the reverse order ^^

Reply:
Since this PR is already merged in, I've called this out in a separate issue (https://github.com/redhat-developer/odo/issues/6283). That would be great if you could take a look. Thanks.